### PR TITLE
WIP: Test/sers can register test

### PR DESCRIPTION
**Pull request by jmarx**

### CODE REVIEW

File: .github/workflows/coding-standards.yml [hunk 1]
```diff
@@ -1,101 +1,101 @@
...
-        run: npm ci --legacy-peer-deps
+            - name: NPM install
+              run: npm ci --legacy-peer-deps
 
-      - name: Run JSHint
-        run: npm run lint:js
-        if: ${{ success() || failure() }}
+            - name: Run JSHint
+              run: npm run lint:js
+              if: ${{ success() || failure() }}
```
Comment (mauteri) [inline .github/workflows/coding-standards.yml:101]: please revert these. not sure why they are changing.

File: .github/workflows/phpunit-tests.yml [hunk 2]
```diff
@@ -1,58 +1,58 @@
...
+
+            - name: Fix Code Coverage Paths
+              run: sed -i 's@'$GITHUB_WORKSPACE'@/github/workspace/@g' coverage.xml
+
+            - name: SonarCloud Scan
+              if: github.ref == 'refs/heads/main'
+              uses: SonarSource/sonarcloud-github-action@master
+              env:
+                  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+                  SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
```
Comment (mauteri) [inline .github/workflows/phpunit-tests.yml:58]: please revert. not sure why this is changing.

File: includes/core/classes/class-setup.php [hunk 3]
```diff
@@ -679,4 +680,29 @@ public function set_event_archive_labels( array $post_states, WP_Post $post ): a
...
+	public function check_users_can_register() : void {
+		if ( filter_var( get_option( 'users_can_register' ), FILTER_VALIDATE_BOOLEAN ) || filter_var( get_option( 'gp_suppress_membership_notification' ), FILTER_VALIDATE_BOOLEAN ) ) {
+			return;
+		}
+
+		if ( null !== filter_input( INPUT_GET, 'action' ) && 'suppress_gp_membership_notification' === filter_input( INPUT_GET, 'action' ) && ! empty( filter_input( INPUT_GET, '_wpnonce' ) ) && wp_verify_nonce( sanitize_text_field( wp_unslash( filter_input( INPUT_GET, '_wpnonce' ) ) ), 'clear-notification' ) ) {
+			update_option( 'gp_suppress_membership_notification', true );
+		} else {
+			Utility::render_template(
+				sprintf( '%s/includes/templates/admin/settings/dismiss-notification.php', GATHERPRESS_CORE_PATH ),
```
Comment (mauteri) [inline includes/core/classes/class-setup.php:701]: this isn't a `setting` put in a `setup` directory: '%s/includes/templates/admin/setup/dismiss-notification.php'

File: includes/core/classes/class-setup.php [hunk 4]
```diff
@@ -679,4 +680,29 @@ public function set_event_archive_labels( array $post_states, WP_Post $post ): a
...
+	 * Display a notification to recommend enabling user registration for GatherPress functionality.
+	 *
+	 * This method checks if user registration is enabled in WordPress settings and displays a
+	 * notification encouraging users to enable registration for optimal GatherPress functionality.
+	 * Users have the option to suppress this notification permanently.
+	 *
+	 * @return void
+	 */
+	public function check_users_can_register() : void {
+		if ( filter_var( get_option( 'users_can_register' ), FILTER_VALIDATE_BOOLEAN ) || filter_var( get_option( 'gp_suppress_membership_notification' ), FILTER_VALIDATE_BOOLEAN ) ) {
```
Comment (mauteri) [inline includes/core/classes/class-setup.php:693]: see comment in paul's PR about lining this up on multiple lines to be more readable.

File: includes/core/classes/class-setup.php [hunk 5]
```diff
@@ -679,4 +680,29 @@ public function set_event_archive_labels( array $post_states, WP_Post $post ): a
...
+			return;
+		}
+
+		if ( null !== filter_input( INPUT_GET, 'action' ) && 'suppress_gp_membership_notification' === filter_input( INPUT_GET, 'action' ) && ! empty( filter_input( INPUT_GET, '_wpnonce' ) ) && wp_verify_nonce( sanitize_text_field( wp_unslash( filter_input( INPUT_GET, '_wpnonce' ) ) ), 'clear-notification' ) ) {
```
Comment (mauteri) [inline includes/core/classes/class-setup.php:697]: see comment in paul's PR about lining this up on multiple lines to be more readable.

File: includes/templates/admin/settings/dismiss-notification.php [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+		<p>
+			<?php
+				/* translators: %2$s: Registration URL, %4$s: Dismiss URL. */
+				printf(
+					' %1$s <a href= %2$s> %3$s</a>. <a href= %4$s><button>Dismiss forever</button></a>',
+					esc_html__( 'To ensure GatherPress functions optimally, we recommend enabling user registration. You can do so by', 'gatherpress' ),
+					esc_url( admin_url( 'options-general.php#users_can_register' ) ),
+					esc_html__( 'enabling user registration here', 'gatherpress' ),
+					esc_url( wp_nonce_url( add_query_arg( 'action', 'suppress_gp_membership_notification' ), 'clear-notification' ) )
+				);
```
Comment (mauteri) [inline includes/templates/admin/settings/dismiss-notification.php:20]: this isn't right. take a look at Otto's post here: http://ottopress.com/2012/internationalization-youre-probably-doing-it-wrong/

Comment (mauteri) [reply includes/templates/admin/settings/dismiss-notification.php:20]: `Dismiss forever` should also be internationalized.

**Review by mauteri — CHANGES_REQUESTED** [review]

File: .github/workflows/coding-standards.yml (hunk 1 above)
Comment (mauteri) [reply .github/workflows/coding-standards.yml:101]: Additional notes on this:

Screenshot 2023-09-10 at 5 37 40 PM

**Review by mauteri — CHANGES_REQUESTED** [review]

File: includes/core/classes/class-setup.php (hunk 3 above)
Comment (jmarx) [reply includes/core/classes/class-setup.php:708]: @mauteri @pbrocks 
Mike, I don't see any setup directory?
Screenshot 2023-09-11 at 8 40 35 AM

File: includes/core/classes/class-setup.php [hunk 7]
```diff
@@ -679,4 +680,36 @@ public function set_event_archive_labels( array $post_states, WP_Post $post ): a
...
+	 * notification encouraging users to enable registration for optimal GatherPress functionality.
+	 * Users have the option to suppress this notification permanently.
+	 *
+	 * @return void
+	 */
+	public function check_users_can_register() : void {
+		if (
+			filter_var( get_option( 'users_can_register' ), FILTER_VALIDATE_BOOLEAN ) ||
+			filter_var( get_option( 'gp_suppress_membership_notification' ),FILTER_VALIDATE_BOOLEAN ) 
+			) {
```
Comment (mauteri) [inline includes/core/classes/class-setup.php:696]: should look like this
```
		if (
			filter_var( get_option( 'users_can_register' ), FILTER_VALIDATE_BOOLEAN ) ||
			filter_var( get_option( 'gp_suppress_membership_notification' ), FILTER_VALIDATE_BOOLEAN ) 
		) {
```

File: includes/core/classes/class-setup.php [hunk 8]
```diff
@@ -679,4 +680,36 @@ public function set_event_archive_labels( array $post_states, WP_Post $post ): a
...
+			return;
+		}
+
+		if (
+			null !== filter_input( INPUT_GET, 'action' ) && 'suppress_gp_membership_notification' === filter_input( INPUT_GET, 'action' ) &&
+			! empty( filter_input( INPUT_GET, '_wpnonce' ) ) &&
+			wp_verify_nonce( sanitize_text_field( wp_unslash( filter_input( INPUT_GET,'_wpnonce' ) ) ), 'clear-notification' ) 
+			) {
```
Comment (mauteri) [inline includes/core/classes/class-setup.php:704]:
```
		if (
			null !== filter_input( INPUT_GET, 'action' ) && 
            'suppress_gp_membership_notification' === filter_input( INPUT_GET, 'action' ) &&
			! empty( filter_input( INPUT_GET, '_wpnonce' ) ) &&
			wp_verify_nonce( sanitize_text_field( wp_unslash( filter_input( INPUT_GET, '_wpnonce' ) ) ), 'clear-notification' ) 
		) {
```

**Review by mauteri — CHANGES_REQUESTED** [review]

File: includes/core/classes/class-setup.php (hunk 3 above)
Comment (jmarx) [reply includes/core/classes/class-setup.php:708]: OK, got it. So you want me to move this file into the folder. I thought you were asking to reference a phantom directory :) @mauteri

**Review by mauteri — APPROVED** [review]

**Pull request merged**